pkg/apis/triggers/v1alpha1/trigger_template_types.go

```diff
@@ -0,0 +1,63 @@
+package v1alpha1
+
+/*
+Copyright 2019 The Tekton Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Check that TriggerTemplate may be validated and defaulted.
+//var _ apis.Validatable = (*TriggerTemplate)(nil)
+
+// +k8s:deepcopy-gen=true
+type TriggerTemplateSpec struct {
+	Params            []Param                   `json:"params,omitempty"`
```

Comment:
i wonder if it would be worth actually importing the `ParamSpec` type from Pipelines and using it directly - would come with its own validation, etc.

Reply:
oh yeah, it def is. thanks!